### PR TITLE
🌱 improve how do we perform issue triage and milestone management

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Proposing a change to how we perform issue triage and milestone management:
- Stop assigning issues to the current milestone as a signal for "triaged", and start using the triage labels instead
- Start adding issues to milestones only when there is someone who expressed intent to work on it before the deadline
- Drop Milestone Next

/hold 
for community discussion

/cc @CecileRobertMichon @enxebre  @vincepri @sbueringer

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

This should be a big improvement for signalling what's coming in a new release - big +1

File: CONTRIBUTING.md
Comment (CecileRobertMichon) [inline CONTRIBUTING.md:188]: > to complete the work before the release date

before the release code freeze target date?

File: CONTRIBUTING.md
Comment (sbueringer) [reply CONTRIBUTING.md:183]: In my opinion we shouldn't wait, except something has been explicitly defined as release-blocking

Comment (CecileRobertMichon) [reply CONTRIBUTING.md:183]: I agree. In that case, it might be worth making this explicit so it's clear to contributors and users that an issue being in the milestone doesn't guarantee inclusion in the release.

Comment (vincepri) [reply CONTRIBUTING.md:183]: +1 to not wait

File: CONTRIBUTING.md
Comment (vincepri) [inline CONTRIBUTING.md:193]: By dropping the milestone Next, are we planning to keep issues without a milestone?

File: CONTRIBUTING.md
Comment (vincepri) [inline CONTRIBUTING.md:277]: +1 on this

File: CONTRIBUTING.md
Comment (sbueringer) [reply CONTRIBUTING.md:193]: That would have been my assumption

File: CONTRIBUTING.md
Comment (sbueringer) [inline CONTRIBUTING.md:184]: Are we planning to do this for every single issue that get's implemented? Or is just something we would do if we want to if a contributor wants to set the expectation that it lands in a release?

Or maybe only for bigger issues?

Sounds like a lot of work/toil for us if we do this for every issue that someone starts working on.

File: CONTRIBUTING.md
Comment (sbueringer) [inline CONTRIBUTING.md:198]: nit: I think based on our current policy it comes down to only N as long as we don't bump the contract (which we didn't do for a while).

Example:
* Right now:
    * v1.0 => out of support
    * v1.1 => patch releases
    * v1.2 is not out yet => no release obviously
* After v1.2 release
    * v1.1. support is immediately dropped based on policy
    * v1.2 => patch releases

File: CONTRIBUTING.md
Comment (sbueringer) [inline CONTRIBUTING.md:274]:
```suggestion
- `triage/duplicate`: Indicates an issue is a duplicate of another open issue. 
```
nit

**Comment by sbueringer** [comment]
Thx for tackling this. I think it's a good step to reduce toil for maintainers

File: CONTRIBUTING.md
Comment (fabriziopandini) [reply CONTRIBUTING.md:193]: Yes, triage accepted, no milestone = milestone next

File: CONTRIBUTING.md
Comment (fabriziopandini) [reply CONTRIBUTING.md:183]: Updated:
- Adding an issue to the release only imply the issue will be tracked
- We should add to milestone/track only major features/bug fixes specifically targeting a release; everything else will simply merge when ready without additional toil
- There is an item calling out that an issue being in the milestone doesn't guarantee inclusion in the release

File: CONTRIBUTING.md
Comment (fabriziopandini) [reply CONTRIBUTING.md:184]: answered in https://github.com/kubernetes-sigs/cluster-api/pull/6828#discussion_r931286434

File: CONTRIBUTING.md
Comment (fabriziopandini) [reply CONTRIBUTING.md:198]: fixed

**Review by vincepri — APPROVED** [review]
/approve

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Comment by fabriziopandini** [comment]
As per 27th July office hours discussion lazy consensus set for EOW (29th of July)

**Comment by fabriziopandini** [comment]
Commit squashed
/hold cancel
As per community meeting discussion

**Review by vincepri — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
Thx for tackling this! Big improvement in my opinion

/lgtm

**Pull request merged**
